### DISCUSSION TITLE
Age of Shadows

### DISCUSSION THREAD

**Discussion by nerun** [modernuo/ModernUO#1455] [Q&A] [answered]
I am trying to run `AOS`. After compile, i choose client folder which is `version 4.0.0m` and it says:

```
$ dotnet Distribution/ModernUO.dll
ModernUO - [https://github.com/modernuo/modernuo] Version 0.9.8.-1
Copyright 2019-2023 ModernUO Development Team
This program comes with ABSOLUTELY NO WARRANTY;
This is free software, and you are welcome to redistribute it under certain conditions.

You should have received a copy of the GNU General Public License
along with this program. If not, see <https://www.gnu.org/licenses/>.

Please enter the absolute path to your ClassicUO or Ultima Online data:
 > ../../ClientAOS
Added ../../ClientAOS.
[enter to finish]> 
Error:
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
   at Server.UOClient.DetectClassicClient() in /data/UO/ModernUO-0.9.8.27/Projects/Server/Client/UOClient.cs:line 92
   at Server.UOClient.Load() in /data/UO/ModernUO-0.9.8.27/Projects/Server/Client/UOClient.cs:line 40
   at Server.ServerConfiguration.Load(Boolean mocked) in /data/UO/ModernUO-0.9.8.27/Projects/Server/Configuration/ServerConfiguration.cs:line 254
   at Server.Core.Main(String[] args) in /data/UO/ModernUO-0.9.8.27/Projects/Server/Main.cs:line 472
This exception is fatal, press return to exit

zsh: killed     dotnet Distribution/ModernUO.dll
```
Also tested `Client 4.0.11c (Samurai Empire)` and got the same issue. But tested with `Client 5.0.9.1 (Mondain's Legacy)` and it works.

Just let me say: [Kamron Batman](https://github.com/kamronbatman) you are my hero! Great emulator dude!

**Comment by kamronbatman** [comment]
Yeah I will look into it. Most likely the way I am detecting the client version doesn't work for such an old client.

**Reply by kamronbatman** [reply]
So I am pulling the version from the file info attributes, which didn't exist prior to client v5. I am not sure where it is in the binary file code.

**Reply by kamronbatman** [reply, marked as answer]
Looks like the earliest client we can detect is 5.0.4d. I'll fix the bug so it returns null and doesn't crash.

Note: I may also need to add a manual conversion for versions < 5.0.7.0 to switch to/from the letter suffix. The file for example says 5.0.5.1 which is 5.0.5b.

**Comment by nerun** [comment]
Thank you man! I will try the new update asap.
